src/main/java/com/epam/aidial/core/controller/DeploymentPostController.java

```diff
@@ -77,37 +81,69 @@ public Future<?> handle(String deploymentId, String deploymentApi) {
         }
 
         Deployment deployment = context.getConfig().selectDeployment(deploymentId);
-        if (!isValidDeploymentApi(deployment, deploymentApi)) {
-            deployment = null;
-        }
+        boolean isValidDeployment = isValidDeploymentApi(deployment, deploymentApi);
 
-        if (deployment == null) {
-            log.error("Deployment {} is not found", deploymentId);
-            return context.respond(HttpStatus.NOT_FOUND, "Deployment is not found");
+        if (!isValidDeployment) {
+            log.warn("Deployment {}/{} is not valid", deploymentId, deploymentApi);
+            return respond(HttpStatus.NOT_FOUND, "Deployment is not found");
         }
 
-        if (!isBaseAssistant(deployment) && !DeploymentController.hasAccess(context, deployment)) {
-            log.error("Forbidden deployment {}. Key: {}. User sub: {}", deploymentId, context.getProject(), context.getUserSub());
-            return context.respond(HttpStatus.FORBIDDEN, "Forbidden deployment");
+        Promise<Void> promise = Promise.promise();
```

Comment:
we can't due to exception handling

Reply:
Please add more details, it is not clear for me.

Reply:
controller must return succeeded future to proceed with our workflow

You propose to return a future from vertx.executeBlocking chain, if this chain fails somewhere - then you send exceptional response here and in proxy twice.

Reply:
The following code should work.
```
.otherwise(error-> {
                    handleRequestError(promise, deploymentId, error);
                    return null;
                })
```